### ISSUE TITLE
CLI prompt to re-run a full build

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#3506]
The ava test runner has a nice option to rerun tests with `r` and it’d be nice to have something similar in Eleventy (especially with incremental builds)

<img width="393" alt="1297 tests passed, 19 tests skipped, Type `r` and press enter to rerun tests, Type `u` and press enter to update snapshots" src="https://github.com/user-attachments/assets/d9ffffa4-7905-4201-aa0c-aace792c3f20">

**Comment by zachleat** [comment]
interesting Node.js pattern here: https://www.linkedin.com/posts/mapocock_til-you-can-prompt-folks-on-your-cli-without-activity-7255214390419894272-SXMi?utm_source=share&utm_medium=member_desktop

<img width="714" alt="image" src="https://github.com/user-attachments/assets/5c507962-08c3-4311-b79f-52b63ef0fe56">

**Comment by Ryuno-Ki** [comment]
```js
import { stdin, stdout } from "node:process";
import { createInterface } from "node:readline/promises";

// 1. Create an interface, taking in stdin and stdout
const rl = createInterface({
  input: stdin,
  output: stdout,
});

// 2. Ask the user a question, and await the result!
const answer = await rl.question(
  "How attractive is Matt Pocock? ",
);

// 3. Get access to the answer immediately
console.log(
  `Thank you for your valuable feedback: ${answer}`,
);

// 4. Close the interface
rl.close();
```

Transcribed, because blind people should be able to learn as well :)

[Node.js API documentation on the subject](https://nodejs.org/api/readline.html#readlinepromisescreateinterfaceoptions)

I would expect some signal handling (<kbd>Ctrl</kbd>+<kbd>C</kbd> and friends). Perhaps best to have a try-catch around the question?

[Node itself has a learning module](https://nodejs.org/en/learn/command-line/accept-input-from-the-command-line-in-nodejs) with some recommendations on handling sensitive data. Keep in mind that the callback version should work better for larger data (because RAM limitations and streaming).

**Comment by Ryuno-Ki** [comment]
Since this is dealing with readline I want to highlight an article by @jvns

https://jvns.ca/blog/2024/07/08/readline/

TL;DR: It can be complicated.
